rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -252,24 +253,17 @@ public class RskContext implements NodeContext, NodeBootstrapper {
     private PeerScoringReporterService peerScoringReporterService;
     private TxQuotaChecker txQuotaChecker;
     private GasPriceTracker gasPriceTracker;
-
     private volatile boolean closed;
 
     /***** Constructors ***********************************************************************************************/
 
-    public RskContext(String[] args) {
-        this(args, false);
-    }
-
-    public RskContext(String[] args, boolean ignoreUnmatchedArgs) {
-        this(new CliArgs.Parser<>(
-                NodeCliOptions.class,
-                NodeCliFlags.class,
-                ignoreUnmatchedArgs
-        ).parse(args));
+    public RskContext(RskCli rskCli) {
+        this(rskCli, rskCli.getCliArgs());
     }
-
-    private RskContext(CliArgs<NodeCliOptions, NodeCliFlags> cliArgs) {
+    // ok, i got rid of rskCli field as its not used,
+    // but maybe there's no need to pass in rskCli and just pass the cliArgs.
+    private RskContext(RskCli rskCli, CliArgs<NodeCliOptions, NodeCliFlags> cliArgs) {
```

Comment:
Your comment is completely true, I don't think you need to store `rskCli` arg here for now.